game/Submods/NSFW Submod/nsfw_fetish.rpy

```diff
@@ -0,0 +1,295 @@
+            action=EV_ACT_QUEUE,
+            aff_range=(mas_aff.LOVE, None)
+        )
+    )
+
+label nsfw_player_fetishintro:
+    m 1eua "Hey, [player]. I just wanted to tell you about some work I've done!"
+    m 1eua "You remember talking with me about fetishes?"
+    m 1eua "...Well I made something so that we could talk about them."
+    m 1eua "I'm not very...{i}experienced{/i} with sexual things like this, so I've been looking into them!"
+    m 1eua "And at the end"
+    extend 2eua "...you get to tell me if you're into it or not."
+    m 1eua "Don't worry! You can change your mind whenever, I don't mind."
+    m 1eua "I'd love to help you explore your fetishes, [player]. Ahaha~"
+    $ mas_unlockEventLabel("nsfw_player_fetishes")
+return
+
+# entry point for fetish flow
+init 6 python: # Use init 6 since the dictionary entry to store our entries is added in 5, and we want it around
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="nsfw_player_fetishes",
+            category=['sex'],
+            prompt="Let's talk about fetishes.",
+            conditional="store.mas_getEVL_shown_count('nsfw_player_fetishintro') >= 1",
+            action=EV_ACT_UNLOCK,
+            aff_range=(mas_aff.LOVE, None)
+        )
+    )
+
+label nsfw_player_fetishes:
+    python:
+        # Unlock any fetish that need to be unlocked
+        Event.checkEvents(nsfw_fetish.nsfw_fetish_database)
+
+        # build menu list
+        nsfw_fetish_menu_items = [
+            (ev.prompt, ev_label, not seen_event(ev_label), False)
+            for ev_label, ev in nsfw_fetish.nsfw_fetish_database.iteritems()
+            if (
+                Event._filterEvent(ev, unlocked=True, aff=mas_curr_affection, flag_ban=EV_FLAG_HFM)
+                and ev.checkConditional()
+            )
+        ]
+
+        # also sort this list
+        nsfw_fetish_menu_items.sort()
+
+        # final quit item
+        final_item = ("Oh nevermind.", False, False, False, 20)
+
+    # move Monika to the left
+    show monika at t21
+
+    # call scrollable pane
+    call screen mas_gen_scrollable_menu(nsfw_fetish_menu_items, mas_ui.SCROLLABLE_MENU_MEDIUM_AREA, mas_ui.SCROLLABLE_MENU_XALIGN, final_item)
+
+    # return value? then push
+    if _return:
+        $ nsfw_fetish.nsfw_fetish_delegate_callback()
+        $ pushEvent(_return)
+        # move her back to center
+        show monika at t11
+
+    else:
+        return "prompt"
+
+    return
+
+# NSFW fetishes start here
+# ---------------------------
+# Noting that the current approach is intended to be both informative and descriptive, allowing you to make a decision of whether or not it's something you're into.
+
+
+default persistant.mas_pm_bondage = False
+default persistant.mas_pm_bondage_context = "U" #G for Giving, R for Recieving, B for both, U for Undefined

```

Comment:
I believe these two are supposed to be 'persistent' vs. 'persistant'.
Also, we want to avoid using the 'mas_' prefix for any of our persistent variable declarations.
Reasoning being the devs want that reserved for official MAS variables. Whilst technically I can never see them having 'bondage' as one of their variables, it's better to be safe than sorry. I've been using the 'nsfw_' prefix since it's the keyword for our submod.